test/effects/effect_test_utils.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:math';
+
+import 'package:flame/anchor.dart';
+import 'package:flame/components/position_component.dart';
+import 'package:flame/effects/effects.dart';
+import 'package:flame/extensions/vector2.dart';
+import 'package:flame/game/base_game.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+final Random random = Random();
+
+class Callback {
+  bool isCalled = false;
+  void call() => isCalled = true;
+}
+
+void effectTest(
+  WidgetTester tester,
+  PositionComponent component,
+  PositionComponentEffect effect, {
+  bool shouldFinish = true,
+  double iterations = 1.0,
+  double expectedAngle = 0.0,
+  Vector2 expectedPosition,
+  Vector2 expectedSize,
+}) async {
+  expectedPosition ??= Vector2.zero();
+  expectedSize ??= Vector2.all(100.0);
+  final Callback callback = Callback();
+  effect.onComplete = callback.call;
+  final BaseGame game = BaseGame();
+  game.add(component);
+  component.addEffect(effect);
+  final double duration = effect.iterationTime;
+  await tester.pumpWidget(game.widget);
+  double timeLeft = iterations * duration;
+  while (timeLeft > 0) {
+    double stepDelta = 50.0 + random.nextInt(50);
+    stepDelta /= 1000;
+    stepDelta = stepDelta < timeLeft ? stepDelta : timeLeft;
+    game.update(stepDelta);
+    timeLeft -= stepDelta;
+  }
+  if (!shouldFinish) {
+    const double floatRange = 0.001;
+    bool acceptableVector(Vector2 vector, Vector2 expectedVector) {
+      return (expectedVector - vector).length < floatRange;
+    }
+
+    final bool acceptablePosition =
+        acceptableVector(component.position, expectedPosition);
+    final bool acceptableSize = acceptableVector(component.size, expectedSize);
+    final bool acceptableAngle =
+        (expectedAngle - component.angle).abs() < floatRange;
```

Comment:
I wonder if dart test thing has a method to do this. most test frameworks have a assertDouble sort of thing with customizable precision

Reply:
That would be great, haven't found anything like it though.

Reply:
what about `https://pub.dev/documentation/matcher/latest/matcher/closeTo.html`?

Reply:
That worked super, thanks!